### ISSUE TITLE
SEP-24: Lumen deposit/withdrawal

### ISSUE THREAD

**Issue by andywer** [stellar/stellar-protocol#443]
We have a couple of use cases on our minds that would require / benefit from the SEP-24 spec being extended to cover the native lumen asset as well.

Potential side effect: The change proposed here might also allow depositing/withdrawing assets that are not issued by that anchor.

## Proposed change

Let's define a standard way to refer to the native XLM asset in the transfer server endpoints. As far as I can see right now, that should be about the only thing blocking the feature right now.

Right now there is only the `asset_code` property which is supposed to denote the asset issued by that anchor that has this very code. Hence, there is no way to reference an asset that is not issued by this anchor.

### Option 1: Add `asset_issuer` and `asset_type`

The data structure could made consistent with the returned by Horizon, allowing the transfer server to reference any Stellar asset.

Would add a bit of bloat to the data structures, though. Might be mitigated by making `asset_issuer` and `asset_type` optional, which would also ensure backwards compatibility.

### Option 2: Magic `asset_code` value

An alternative approach might be to define a special value that indicates that we mean the native XLM asset. Such a value might be `$XLM` or similar.

It avoids adding any new props, but is less versatile, as you would not be able to reference any non-native assets issued by other anchors.

## Motivation

One motivation for this feature is to extend SEP-24 so that it serves as an open API standard for lumen onramps (like Coinbase or Moonpay). Another one is to allow offramps where you can pay in XLM, USD, EURT, ..., but without the need to issue yet another USD, EUR token.

Happy to hear your thoughts! :)

**Comment by leighmcculloch** [comment]
[SEP-11](https://github.com/stellar/stellar-protocol/blob/master/ecosystem/sep-0011.md) contains details about how we should render assets and the native asset to string, using `"<code>:<issuer>"` for assets, and `"native"` for the native asset and we use that format in Horizon. @msfeldstein Maybe we should use this common format described in SEP-11 and used in Horizon in this SEP also?

In Ticker which requires a code we use `XLM` paired with an issuer of `native`. Maybe we could use that same pattern here if we absolutely need to express the code and issuer separately.

**Comment by msfeldstein** [comment]
So the complication here is that asset_code in `deposit` and `withdraw` are real-world assets, not the issued token, so adding asset_issuer doesn't necessarily make sense.  [`Ex if a user deposits BTC and receives MyBTC tokens, asset_code must be BTC.`](https://github.com/stellar/stellar-protocol/blob/master/ecosystem/sep-0024.md#request)

It seems more like we want a `requested_asset_type`, so you would deposit `USD` and request `XLM` to show up in your account, am i understanding this correctly?

**Comment by andywer** [comment]
@msfeldstein I think we have a bit of an issue, since depositing and withdrawing are not quite uniform: You referenced the depositing and there it is completely correct what you said. For withdrawal, however, it says [Code of the asset the user wants to withdraw. This must match the asset code issued by the anchor. Ex if a user withdraws MyBTC tokens and receives BTC, the asset_code must be MyBTC](https://github.com/stellar/stellar-protocol/blob/master/ecosystem/sep-0024.md#request-1).

Maybe we need to think about how to make the interpretation of those asset references more homogenous first? (Seriously asking, I am still pondering about it myself 😉)

**Comment by msfeldstein** [comment]
Yeah i don't like that distinction either, i'd rather both sides reference the stellar asset in question.  We might be able to knock out both problems with one change (though that would be a breaking one)

**Comment by andywer** [comment]
Sorry for the long pause here, we have just been suuuuper busy.

Short recap:
- we should probably unify the interpretation of the `asset_code` on deposit and withdrawal (breaking change!)
- and/or split the asset field in two: the asset you wish to send and the one you expect to receive on the other side (use case: buy XLM for USD on coinbase.com via SEP-24, for instance)
- denoting assets not issued by this anchor: [SEP-11-like or Ticker API alike](https://github.com/stellar/stellar-protocol/issues/443#issuecomment-547996301)

**Comment by tomerweller** [comment]
I think we should converge on using Stellar asset codes and their SEP11 representation. 

Splitting the asset code to the stellar side and the "real world" side can put us in a world of pain. There are many on-ramps that support multiple currencies or different currency limits in different regions (the big exchanges are like that) and we don't really want to go there. A user should care about coinbase selling XLM, but less so about which currency Coinbase is going to quote me in.

**Comment by msfeldstein** [comment]
I created a PR to see what this would look like if we added asset_issuer.  I think this causes less churn than switching to SEP-11 style asset representation.  We don't need to choreograph a migration this way, since there is a default value (their own issued token) when asset_issuer isn't there.

This also changes the deposit meaning of asset_code to match the withdraw meaning, in that they both reference the stellar asset code.  This could cause some issues for existing anchors who would need to update their implementation when their stellar asset code doesn't match the real world asset code
